core-plugins/src/main/java/co/cask/hydrator/plugin/batch/sink/S3BatchSink.java

```diff
@@ -60,8 +61,12 @@ protected S3BatchSink(S3BatchSinkConfig config) {
     this.config = config;
     // update fileSystemProperties to include accessID and accessKey, so prepareRun can only set fileSystemProperties
     // in configuration, and not deal with accessID and accessKey separately
-    this.config.fileSystemProperties = updateFileSystemProperties(this.config.fileSystemProperties,
-                                                                  this.config.accessID, this.config.accessKey);
+    // do not create file system properties if macros were provided, wait for runtime
+    if (!this.config.containsMacro("fileSystemProperties") && !this.config.containsMacro("accessID") &&
+        !this.config.containsMacro("accessKey")) {
+      this.config.fileSystemProperties = updateFileSystemProperties(this.config.fileSystemProperties,
```

Comment:
the constructor is called only during test cases right, do you need this for test case fix?